src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1799,6 +1805,34 @@ bool core::check_tx_semantic(const transaction& tx, bool keeped_by_block) const
     return true;
 }
 //-----------------------------------------------------------------------------------------------
+void core::check_service_node_ip_address() {
+    if (!m_service_node || m_has_ip_check_disabled || service_node_list.debug_allow_local_ips) {
+        return;
+    }
+
+    auto service_node_ip = epee::string_tools::get_ip_string_from_int32(m_sn_public_ip);
+    // NOTE - this connection won't work as intended if oxenmq's incomplete `SN_ADDR_SELF`
+    // gets implemented.
```

Comment:
This isn't the case anymore (that was only potentially applicable when connecting via pubkey lookup).
```suggestion
```